ui/src/main/resources/FileManagerCode/DriveSheet.xml

```diff
@@ -1832,7 +1832,15 @@ require(['jquery', 'angular', 'JobRunner', 'angular-resource', 'angular-ui-boots
           if (window.confirm("$services.localization.render('fileManager.action.deleteFilesConfirmation')")) {
             // Clear the selection.
             scope.selection = {};
-            scope.drive['delete'](getPaths(files)).done(refreshLiveTable);
+            scope.drive['delete'](getPaths(files)).done(function(result) {
+              if(result.error) {
+                new XWiki.widgets.Notification(
+                  '$escapetool.json($services.localization.render("fileManager.action.deleteError"))'
```

Comment:
Thanks. Fixed.
paul

Reply:
> That means that the localization is invoked at JS-time.

No. The localization is invoked from Velocity with the ``__error__`` parameter which you then replace from JavaScript with the actual error message. But the translation message should have a parameter.

Reply:
> I meant here that using {0} would mean that we do localization at JS time whereas using error allows the replacement to be done at JS time but localization to be done at velocity time.

No. First of all, it doesn't matter where the localization message is used, from Java, JavaScript, Velocity or whatever language. The localization message is **programming language agnostic**. In this case the localization message has a parameter, the error message, so it should look like this:

    fileManager.action.deleteError=An error occurred while deleting ({0}).

Next, since:
* you can't access the localization message directly from JavaScript (you have to use Velocity for this)
* but the parameter expected by the localization message is available only in JavaScript

you need to do this:

```
var message = $jsontool.serialize($services.localization.render('fileManager.action.deleteError', ['__error__']));
new XWiki.widgets.Notification(message.replace('__error__', result.error), 'error');
```

The advantage of this is that:
* the person handling the translations doesn't have to know how the translation is used or what ``__error__`` means; they just have to know that the translation has a parameter
* the ``__error__`` placeholder is added and replaced on consecutive lines of code which makes it easy to understand its need. Otherwise, someone that looks at the translation file after 1 year won't understand why it contains ``__error__`` and why it's not translated like the rest of the text.

Reply:
Fixed in aafcdd8cd23bdc14ea48658cf50e20c9da0a5fc4.
thanks
paul